src/app/task-details/task-details.component.ts

```diff
@@ -14,14 +14,20 @@ import { TaskService } from '../task.service';
 })
 
 export class TaskDetailsComponent implements OnInit {
-  constructor(private route: ActivatedRoute, private taskService: TaskService) { };
-
-  taskList = TaskJson;
   curTask: TaskData | null = null;
 
+  constructor(private route: ActivatedRoute, private taskService: TaskService) { }
+
   ngOnInit() {
-    var id = this.route.snapshot.params['id'];
-    this.curTask = this.taskService.getTaskByID(id);
+    const id = this.route.snapshot.params['id'];
+    this.taskService.getTaskByID(id).subscribe({
+      next: (task) => {
+        this.curTask = task;
+      },
+      error: (error) => {
+        console.error('Error fetching task:', error);
+        this.curTask = null;
```

Comment:
since current task is initially null you dont have to reassign null to it if the request fails for some reason